File: backend/entity/feed_schedule.go
package entity

import "time"

// FeedUpdateSchedule เก็บ config ตารางเวลา auto-update สำหรับแต่ละ feed type
type FeedUpdateSchedule struct {
	ID         uint       `gorm:"primaryKey;autoIncrement"`
	FeedType   string     `gorm:"column:feed_type;uniqueIndex;not null"` // "openvas" | "kev" | "epss"
	Frequency  string     `gorm:"column:frequency;not null;default:'daily'"` // daily | monthly | yearly
	Hour       int        `gorm:"column:hour;not null;default:2"`             // 0-23
	Minute     int        `gorm:"column:minute;not null;default:0"`           // 0-59
	DayOfMonth int        `gorm:"column:day_of_month;not null;default:1"`     // 1-31 (for monthly)
	Month      int        `gorm:"column:month;not null;default:1"`            // 1-12 (for yearly)
	Day        int        `gorm:"column:day;not null;default:1"`              // 1-31 (for yearly)
	Enabled    bool       `gorm:"column:enabled;default:true"`
	LastRunAt  *time.Time `gorm:"column:last_run_at"`
	NextRunAt  *time.Time `gorm:"column:next_run_at"`
	CreatedAt  time.Time
	UpdatedAt  time.Time
}

func (FeedUpdateSchedule) TableName() string {
	return "feed_update_schedules"
}

File: frontend/src/services/feedschedule.ts
import axios from "axios";
import { apiUrl, installMaintenanceInterceptor } from "./api";

const api = axios.create({
  baseURL: apiUrl,
  withCredentials: true,
  timeout: 30000,
  headers: { "Content-Type": "application/json", "ngrok-skip-browser-warning": "true" },
});

installMaintenanceInterceptor(api);

// ── Types ──────────────────────────────────────────────────────────────────

export type FeedType = "openvas" | "kev" | "epss" | "exploit";
export type FeedFrequency = "daily" | "monthly" | "yearly";

export type FeedScheduleDTO = {
  id: number;
  feed_type: FeedType;
  frequency: FeedFrequency;
  hour: number;       // 0-23
  minute: number;     // 0-59
  day_of_month: number; // 1-31  (monthly)
  month: number;        // 1-12  (yearly)
  day: number;          // 1-31  (yearly)
  enabled: boolean;
  last_run_at?: string;
  next_run_at?: string;
};

export type UpdateFeedScheduleRequest = {
  frequency?: FeedFrequency;
  hour?: number;
  minute?: number;
  day_of_month?: number;
  month?: number;
  day?: number;
  enabled?: boolean;
};

// ── API ─────────────────────────────────────────────────────────────────────

export const ListFeedSchedules = async (): Promise<FeedScheduleDTO[]> => {
  try {
    const res = await api.get("/feed-schedules");
    const data = res.data?.data ?? res.data;
    return Array.isArray(data) ? (data as FeedScheduleDTO[]) : [];
  } catch (e) {
    console.error("ListFeedSchedules error:", e);
    return [];
  }
};

export const UpdateFeedSchedule = async (
  feedType: FeedType,
  req: UpdateFeedScheduleRequest,
): Promise<FeedScheduleDTO> => {
  const res = await api.put(`/feed-schedules/${feedType}`, req);
  return res.data as FeedScheduleDTO;
};

export const TriggerFeedNow = async (feedType: FeedType): Promise<void> => {
  await api.post(`/feed-schedules/${feedType}/trigger`);
};
